servo/runner.py

```diff
@@ -234,7 +271,13 @@ def _reraise_if_necessary(task: asyncio.Task) -> None:
         self._main_loop_task = asyncio.create_task(self.main_loop(), name=f"main loop for servo {self.optimizer.id}")
         self._main_loop_task.add_done_callback(_reraise_if_necessary)
 
-    async def run(self, *, poll: bool = True) -> None:
+        if diagnostics:
+            self._diagnostics_loop_task = asyncio.create_task(self.diagnostics_check(), name=f"diagnostics for servo {self.optimizer.id}")
+            self._diagnostics_loop_task.add_done_callback(_reraise_if_necessary)
```

Comment:
Diagnostics task should log errors and not re-raise (please add exponential backoff and/or self-termination if there are repeated errors to prevent logs from getting filled with repeat errors)